feat(profiling): add partial rendering of differential flamegraph

JonasBa opened the pull request:
First part of rendering a full size differential flamegraph (same as what we do with flamecharts).

The view is still missing minimap and call tree table rendering

JonasBa committed:
feat(profiling): add partial rendering of differential flamegraph

JonasBa commented on static/app/utils/profiling/hooks/useDifferentialFlamegraphModel.tsx:
```diff
@@ -0,0 +1,93 @@
+import {useMemo} from 'react';
```
This would have been duplicated, so I created a reusable hook. The idea is to pass it a raw before/after response and generate the differential flamegraph

Zylphrex replied:
Could also consider making this a regular helper function and let the caller wrap it in `useMemo`.  But if you want more granular control over how things are memoed, then this is fine too 👍

Zylphrex commented on src/sentry/conf/server.py:
```diff
@@ -1768,6 +1768,8 @@ def custom_parameter_sort(parameter: dict) -> tuple[str, int]:
     "organizations:profiling-cpu-chart": False,
     # Enables differential flamegraph in profiling
     "organizations:profiling-differential-flamegraph": False,
+    # Enables separate differential flamegraph page
+    "organizations:profiling-differential-flamegraph-page": False,
```
Is this from a bad merge somehow? I see this is on master already.

Zylphrex commented on src/sentry/features/__init__.py:
```diff
@@ -182,6 +182,7 @@
 default_manager.add("organizations:profiling-ui-frames", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
 default_manager.add("organizations:profiling-using-transactions", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
 default_manager.add("organizations:profiling-differential-flamegraph", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
+default_manager.add("organizations:profiling-differential-flamegraph-page", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
```
Same here

JonasBa committed:
feat(profiling): split canvas renderer and add minimap (#61715)

Split the differential flamegraph from the one used in the issues list
as we will be rendering more data and add minimap rendering to the
screen

JonasBa committed:
Merge branch 'master' into jb/feat/differential-flamegraph-full-view

JonasBa committed:
feat(profiling): change feature flag
